src/common/buffer.cc

```diff
@@ -676,25 +676,25 @@ using namespace ceph;
 
   const char *buffer::ptr::c_str() const {
     ceph_assert(_raw);
-    if (buffer_track_c_str)
+    if (unlikely(buffer_track_c_str))
```

Comment:
Do we really need `buffer_track_c_str`?

Reply:
do you see it being used anywhere? i'm not finding any callers of buffer::get_c_str_accesses()

Reply:
We might have similar issues with `buffer_track_alloc` and, generally speaking, the whole `CEPH_BUFFER_TRACK`. Investigating and maybe hinting/dropping them is TODO.

Reply:
> do you see it being used anywhere? i'm not finding any callers of buffer::get_c_str_accesses()

I guess this might be purely for debug. If so, keeping the stuff but only for `!NDEBUG` builds is another option.